### DISCUSSION TITLE
How do I check snapshot report not missing on CI

### DISCUSSION THREAD

**Discussion by fisker** [avajs/ava#2754] [Q&A]
When snapshot report (`.md` file) didn't commit, the test script still pass, I was going to use `ava -u && git diff --exit-code` to make sure report won't change on CI. But `-u` seems can't run on CI.

**Comment by novemberborn** [comment]
The report is for humans (and code review diffs). It's probably easiest to write a separate script that looks for `.snap` files and ensures the report is next to it.

**Reply by fisker** [reply]
How do I know it's correct? Eg, updated test, but not updated `.md`. It already exists, but outdated.

**Reply by novemberborn** [reply]
Fair point.

I think with AVA 4 we can recreate the Markdown file from the binary snapshot, but there still might be some edge case scenarios where this results in a different file.

I'm not yet sold on whether AVA should perform these checks, but perhaps to start with we could expose some of this behavior so a different project / script can do it?

**Reply by fisker** [reply]
I don't know, maybe I'll use `AVA_FORCE_CI` to allow snapshot update.

**Reply by novemberborn** [reply]
Now you're just tempting me to make that variable harder to use…

**Reply by fisker** [reply]
I like how jest assert snapshots.
IMHO, people should only care about serialized data(the md file), I don'tcare the raw data is `1` or `"1"`, if I do care I can just snapshot `JSON.stringify(data)`

**Reply by fisker** [reply]
Why not consider change `concordance` change to be just a default serializer, people can serialize whatever he want.
